### PR TITLE
Navigator: stabilize and export APIs

### PR DESCRIPTION
**Pull request by ciampo**
## What?

Part of #59418
Supersedes #60927
Follow-up to #63317

Export the `Navigator` component, its subcomponents, and the `useNavigator` hook as stable APIs from the `@wordpress/components` package

## Why?

This is part of a [larger effort](https://github.com/WordPress/gutenberg/issues/59418) to remove `__experimental` prefix from all "experimental" components, effectively promoting them to regular stable components. See the related issue for more context.

## How?

- Aside from the existing legacy exports, which use the `__experimental` prefix, add new, unprefixed exports:
  - Legacy exports stay untouched;
  - The new, stable exports will adopt slightly different names compared to legacy and the overloaded naming convention that we agreed to apply for component components:
    - `NavigatorProvider` => `Navigator`
    - `NavigatorScreen` => `Navigator.Screen`
    - `NavigatorButton` => `Navigator.Button`
    - `NavigatorBackButton` => `Navigator.BackButton`
    - `NavigatorToParentButton` won't be exported since it's already deprecated in the experimental version of the component
- Updated all JSDocs and README
- Updated Storybook and Unit Tests to use the stable version of the component
- Added Storybook redirect to avoid getting broken links
- A few smaller changes:
  - Updated the internal context namespacing (which shouldn't be a breaking change since context is an internal API of the `@wordpress/components` package)

## Next steps?

- refactor existing usages in Gutenberg to stable version
- deprecate experimental version (both in JSDocs and with a runtime warning) (+ add unit tests for deprecation warnings)

## Testing Instructions

- Technically there shouldn't be any runtime (behavioral) changes to `Navigator`;
- Make sure that Storybook examples are documented correctly and work as expected;
- Smoke test `Navigator` usages across the editor;
- Review JSDocs, READMEs, and IntelliSense snippets for both legacy and stable APIs

## ✍️  Dev note

The legacy set of `__experimentalNavigator*` APIs is deprecated and should instead be imported as `Navigator`. All of the sub-components are also available via the `Navigator` namespace.

Moreover, the `__experimentalNavigatorToParentButton` component and the `goToParent` method available via the `__experimentalUseNavigator` hook are now deprecated, and they now behave identically to the `__experimentalNavigatorBackButton` and the `goBack` method.

To recap:

- `__experimentalNavigatorProvider` => `Navigator`
- `__experimentalNavigatorScreen` => `Navigator.Screen`
-  `__experimentalNavigatorButton` => `Navigator.Button`
- `__experimentalNavigatorBackButton` => `Navigator.BackButton`
- `__experimentalNavigatorToParentButton` => `Navigator.BackButton`
- `__experimentalUseNavigator` => `useNavigator`

### CODE REVIEW

File: packages/components/src/navigator/legacy.ts
Comment (ciampo) [inline packages/components/src/navigator/legacy.ts:1]: Legacy exports have been moved to a separate `legacy.ts` file (including the legacy JSDocs mentioning the components with their experimental export name).

This makes it easy to have separate names from the stable version, different JSDocs, and deprecation warnings.

File: packages/components/src/navigator/index.ts
Comment (ciampo) [inline packages/components/src/navigator/index.ts:1]: New exports (and their up-to-date JSDocs) have been moved to the folder's main `index.ts` file

File: packages/components/src/navigator/navigator-back-button/README.md
Comment (ciampo) [inline packages/components/src/navigator/navigator-back-button/README.md:1]: All READMEs have also been updated:
- removed experimental snippet
- updated the naming of all `Navigator` sub-components, including internal links to other READMEs

File: packages/components/src/navigator/navigator-provider/index.ts
Comment (ciampo) [inline packages/components/src/navigator/navigator-provider/index.ts:1]: This file was actually renamed to `packages/components/src/navigator/navigator/index.tsx`, but GitHub is not picking that up

File: packages/components/src/navigator/navigator/component.tsx [hunk 1]
```diff
@@ -273,41 +273,31 @@ function UnconnectedNavigatorProvider(
 }
 
 /**
- * The `NavigatorProvider` component allows rendering nested views/panels/menus
- * (via the `NavigatorScreen` component and navigate between these different
- * view (via the `NavigatorButton` and `NavigatorBackButton` components or the
+ * The `Navigator` component allows rendering nested views/panels/menus
```
Comment (ciampo) [inline packages/components/src/navigator/navigator/component.tsx]: The top level `Navigator` component (previously named `NavigatorProvider`) is the only component for which I've kept a JSDoc for the internal implementation, because apparently Storybook uses it when displaying docs 🤷  Not sure why it doesn't use the one defined in `packages/components/src/navigator/index.ts`

File: packages/components/src/navigator/stories/index.story.tsx [hunk 2]
```diff
@@ -10,20 +10,20 @@ import Button from '../../button';
...
+	component: Navigator,
+	subcomponents: {
+		// @ts-expect-error - See https://github.com/storybookjs/storybook/issues/23170
+		Screen: Navigator.Screen,
+		// @ts-expect-error - See https://github.com/storybookjs/storybook/issues/23170
+		Button: Navigator.Button,
+		// @ts-expect-error - See https://github.com/storybookjs/storybook/issues/23170
+		BackButton: Navigator.BackButton,
+	},
+	title: 'Components/Navigator',
```
Comment (ciampo) [inline packages/components/src/navigator/stories/index.story.tsx:24]: Updated the `PREVIOUSLY_EXPERIMENTAL_COMPONENTS` in `storybook/manager-head.html` to make sure old links still work

File: packages/components/src/navigator/navigator/component.tsx (hunk 1 above)
Comment (ciampo) [reply packages/components/src/navigator/navigator/component.tsx]: it seems like the `ts` extension was the problem — changing it to `.tsx` seems to allow Storybook to pick the JSDoc up 😅  https://github.com/WordPress/gutenberg/pull/64613/commits/85471a9224994d8977221fc4915a159622c3f307


cc @mirka

File: docs/manifest.json [hunk 3]
```diff
@@ -1103,22 +1103,16 @@
 		"markdown_source": "../packages/components/src/navigator/navigator-button/README.md",
 		"parent": "components"
 	},
-	{
-		"title": "NavigatorProvider",
-		"slug": "navigator-provider",
-		"markdown_source": "../packages/components/src/navigator/navigator-provider/README.md",
-		"parent": "components"
-	},
```
Comment (mirka) [inline docs/manifest.json:1117]: FYI, removing things from the manifest is not going to remove it from the [Block Editor Handbook](https://developer.wordpress.org/block-editor/reference-guides/components/) 😱 There are some extra steps, see https://github.com/WordPress/gutenberg/pull/60003#issuecomment-2010288927

File: packages/components/CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+### New Features
```
Comment (mirka) [inline packages/components/CHANGELOG.md]: Not "Enhancements"? (Enhancement == new feature)

File: packages/components/src/navigator/index.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+ *       <Navigator.BackButton>
+ *         Go back
+ *       </Navigator.BackButton>
+ *     </Navigator.Screen>
+ *   </Navigator>
+ * );
+ * ```
+ */
+export const Navigator = Object.assign( TopLevelNavigator, {
+	dislayName: 'Navigator',
```
Comment (mirka) [inline packages/components/src/navigator/index.tsx]:
```suggestion
```

I can confirm this is unnecessary because it's misspelled 😄

File: packages/components/src/navigator/index.tsx [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+	 *         Go back
+	 *       </Navigator.BackButton>
+	 *     </Navigator.Screen>
+	 *   </Navigator>
+	 * );
+	 * ```
+	 */
+	Screen: Object.assign( NavigatorScreen, {
+		displayName: 'Navigator.Screen',
+	} ),
```
Comment (mirka) [inline packages/components/src/navigator/index.tsx:68]: I actually like this better than doing `NavigatorScreen.displayName = 'Navigator.Screen'` in the individual subcomponent files. It makes more sense to handle that concern in this top-level index file. Should we update the contributing guide accordingly?

File: packages/components/src/navigator/index.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+import { NavigatorBackButton } from './navigator-back-button';
+export { useNavigator } from './use-navigator';
+
+/**
+ * The `Navigator` component allows rendering nested views/panels/menus
+ * (via the `Navigator.Screen` component) and navigate between these different
+ * view (via the `Navigator.Button` and `Navigator.BackButton` components or the
+ * `useNavigator` hook).
+ *
+ * @example
```
Comment (mirka) [inline packages/components/src/navigator/index.tsx]:
```suggestion
```

Should we remove this `@example` tag so the snippet shows up in the Storybook?

File: packages/components/src/navigator/navigator-back-button/component.tsx [hunk 8]
```diff
@@ -21,43 +21,7 @@ function UnconnectedNavigatorBackButton(
...
- *     </NavigatorScreen>
- *   </NavigatorProvider>
- * );
- * ```
- */
 export const NavigatorBackButton = contextConnect(
 	UnconnectedNavigatorBackButton,
-	'NavigatorBackButton'
+	'Navigator.BackButton'
 );
```
Comment (mirka) [inline packages/components/src/navigator/navigator-back-button/component.tsx:27]: Hmm, this is an interesting one. If we do it this way, the context name is going to change for consumers of the legacy `NavigatorBackButton`, right? I wonder if we need separate context-connnected exports for the current and the legacy?

File: packages/components/src/navigator/test/index.tsx [hunk 9]
```diff
@@ -14,14 +14,14 @@ import { useState } from '@wordpress/element';
...
+import { Navigator, useNavigator } from '..';
 import {
-	NavigatorProvider,
-	NavigatorScreen,
-	NavigatorButton,
-	NavigatorBackButton,
+	// NavigatorProvider,
+	// NavigatorScreen,
+	// NavigatorButton,
+	// NavigatorBackButton,
```
Comment (mirka) [inline packages/components/src/navigator/test/index.tsx]: 🧹 ?

File: docs/manifest.json (hunk 3 above)
Comment (ciampo) [reply docs/manifest.json:1117]: Thank you! I'll follow-up accordingly

File: packages/components/CHANGELOG.md (hunk 4 above)
Comment (ciampo) [reply packages/components/CHANGELOG.md]: My reasoning was "New API" (stable `Navigator`) => "New feature" section

Although I'm very much fine putting it under "Enhancements"

File: packages/components/src/navigator/index.tsx (hunk 5 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx]: Lol. That's one way to test changes out

File: packages/components/src/navigator/index.tsx (hunk 6 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx:68]: Sounds good!

File: packages/components/src/navigator/navigator-back-button/component.tsx (hunk 8 above)
Comment (ciampo) [reply packages/components/src/navigator/navigator-back-button/component.tsx:27]: It is, yeah. although I didn't think it would be a problem — from the PR description:

> Updated the internal context namespacing (which shouldn't be a breaking change since context is an internal API of the `@wordpress/components` package)

The approach that you propose probably won't be enough, since we'd need to also change the `useContextSystem` call in the component render functions.

**Comment by ciampo** [comment]
I'm going to resume work on this PR after https://github.com/WordPress/gutenberg/pull/64675 is merged.

File: packages/components/src/navigator/navigator-back-button/component.tsx (hunk 8 above)
Comment (mirka) [reply packages/components/src/navigator/navigator-back-button/component.tsx:27]: Wow, you're right. I always assumed the context API was public! 😂

**Comment by tyxla** [comment]
> I'm going to resume work on this PR after #64675 is merged.

Sounds good. I was starting to review it, but I will halt it for now.

I wonder if it's a good idea to approach this modularly and split whatever we can to separate PRs to keep this one smaller.

File: packages/components/CHANGELOG.md (hunk 4 above)
Comment (ciampo) [reply packages/components/CHANGELOG.md]: Done https://github.com/WordPress/gutenberg/pull/64613/commits/af97de9356dd7a8ba9a3ceeef705edebc0d36e71

File: packages/components/src/navigator/index.tsx (hunk 5 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx]: Done https://github.com/WordPress/gutenberg/pull/64613/commits/81931212710d2112376156173907acaa10d7158a

File: packages/components/src/navigator/index.tsx (hunk 7 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx]: https://github.com/WordPress/gutenberg/pull/64613/commits/e7d20195ce125e78a2594874445bb34cb939766d

File: packages/components/src/navigator/test/index.tsx (hunk 9 above)
Comment (ciampo) [reply packages/components/src/navigator/test/index.tsx]: 🧹  https://github.com/WordPress/gutenberg/pull/64613/commits/0f2d077905f721ecb89023192f16656b4b08fccc

**Comment by ciampo** [comment]
I rebased and addressed simple feedback.

I would still wait before reviewing this PR. The current plan of action is:

- [x] Review and merge https://github.com/WordPress/gutenberg/pull/64786
- [x] Review and merge https://github.com/WordPress/gutenberg/pull/64777
- [x] Rebase this PR and prepare for final review

I've also updated the [tracking issue](https://github.com/WordPress/gutenberg/issues/59418) adding all of the planned follow-up step to this PR.

> I wonder if it's a good idea to approach this modularly and split whatever we can to separate PRs to keep this one smaller.

@tyxla , Even if the size of the PR is medium-large, I believe 99% of the changes left in this PR are simple renames — and in that sense I feel ok with the current size. Having said that, I'll re-assess once all dependency PRs are merged, before asking for a final round of review.

File: packages/components/src/navigator/index.tsx (hunk 6 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx:68]: https://github.com/WordPress/gutenberg/pull/64793

**Comment by tyxla** [comment]
> @tyxla , Even if the size of the PR is medium-large, I believe 99% of the changes left in this PR are simple renames — and in that sense I feel ok with the current size. Having said that, I'll re-assess once all dependency PRs are merged, before asking for a final round of review.

That makes sense, @ciampo, will help with the review once all the dependencies are merged 👍

**Comment by ciampo** [comment]
Rebased on top of trunk to include the latest changes.



> > I wonder if it's a good idea to approach this modularly and split whatever we can to separate PRs to keep this one smaller.
> 
> @tyxla , Even if the size of the PR is medium-large, I believe 99% of the changes left in this PR are simple renames — and in that sense I feel ok with the current size. Having said that, I'll re-assess once all dependency PRs are merged, before asking for a final round of review.

I split some code changes to #65671, which is ready for review.

**Comment by ciampo** [comment]
Rebased to include changes from https://github.com/WordPress/gutenberg/pull/65671

This PR is ready for a final round of review @tyxla @mirka

File: docs/manifest.json (hunk 3 above)
Comment (tyxla) [reply docs/manifest.json:1117]: Just a reminder we'll need to take care of this one after merging.

File: packages/components/CHANGELOG.md [hunk 10]
```diff
@@ -17,6 +17,7 @@
 -   `MenuGroup`: Simplify the MenuGroup styles within dropdown menus. ([#65561](https://github.com/WordPress/gutenberg/pull/65561)).
 -   `DatePicker`: Use compact button size. ([#65653](https://github.com/WordPress/gutenberg/pull/65653)).
 -   `Navigator`: add support for exit animation ([#64777](https://github.com/WordPress/gutenberg/pull/64777)).
+-   `Navigator`: stabilize APIs ([#64613](https://github.com/WordPress/gutenberg/pull/64613)).
```
Comment (tyxla) [inline packages/components/CHANGELOG.md:22]: I think it makes sense to list it under "New features" as well, now that it's stable

File: packages/components/src/navigator/index.tsx [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+	 * );
+	 * ```
+	 */
+	Screen: Object.assign( NavigatorScreen, {
+		displayName: 'Navigator.Screen',
+	} ),
+	/**
+	 * The `Navigator.Button` component can be used to navigate to a screen and
+	 * should be used in combination with the `Navigator`, the `Navigator.Screen`
+	 * and the `Navigator.BackButton` components, and the `useNavigator` hook.
```
Comment (tyxla) [inline packages/components/src/navigator/index.tsx]: We claim that these components should be used also with `useNavigator`, but then should we include the hook usage in the examples?

File: docs/manifest.json [hunk 12]
```diff
@@ -1109,22 +1109,16 @@
 		"markdown_source": "../packages/components/src/navigator/navigator-button/README.md",
```
Comment (tyxla) [inline docs/manifest.json]: I see we continue calling these for example `NavigatorButton`, but they're actually exposed as `Navigator.Button`? Should this be reflected in the manifest? Should we just leave the single `Navigator` and have all READMEs in it?

File: packages/components/src/index.ts [hunk 13]
```diff
@@ -132,6 +132,7 @@ export {
 	NavigatorToParentButton as __experimentalNavigatorToParentButton,
 	useNavigator as __experimentalUseNavigator,
```
Comment (tyxla) [inline packages/components/src/index.ts:133]: Are we planning to deprecate the legacy ones at some point? Perhaps in a separate PR after we update the existing usages?

File: packages/components/src/navigator/navigator-back-button/component.tsx [hunk 14]
```diff
@@ -23,5 +23,5 @@ function UnconnectedNavigatorBackButton(
 
 export const NavigatorBackButton = contextConnect(
 	UnconnectedNavigatorBackButton,
-	'NavigatorBackButton'
+	'Navigator.BackButton'
```
Comment (tyxla) [inline packages/components/src/navigator/navigator-back-button/component.tsx:26]: Double-checking if this is OK - I found it unexpected that here we use the `Navigator.BackButton` context name, but in the button hook, we use `NavigatorBackButton` without the dot notation:

https://github.com/WordPress/gutenberg/blob/fdff00c9b442dd4263da62a7dd8a3f9fe617b3ad/packages/components/src/navigator/navigator-back-button/hook.ts#L23

Same for the rest of the subcomponents and their context.

**Review by tyxla — COMMENTED** [review]
This looks good and tests well to me 👍 Great work @ciampo 🚀 

I've left a few questions before shipping, most are just for confirmation of my assumptions.

File: packages/components/CHANGELOG.md (hunk 10 above)
Comment (ciampo) [reply packages/components/CHANGELOG.md:22]: We had already [discussed](https://github.com/WordPress/gutenberg/pull/64613#discussion_r1726677282) which CHANGELOG section to put this under, and we went for Enhancements (also, similar PRs also added an entry under Enhancements)

Comment (tyxla) [reply packages/components/CHANGELOG.md:22]: I just think it makes sense since it's part of the formalization of the stable API, but I have no strong feelings about it. Leaving to your judgment.

File: packages/components/src/navigator/index.tsx (hunk 11 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx]: Good point. TBH, `useNavigator` is more of an advanced usage (and there's an example of it in Storybook). Do you think that's enough, or should I look into making the snippet longer and incorporate `useNavigator`?

File: packages/components/src/index.ts (hunk 13 above)
Comment (ciampo) [reply packages/components/src/index.ts:133]: Yup! From the "Next steps" section in the PR description:

> - refactor existing usages in Gutenberg to stable version
> - deprecate experimental version (both in JSDocs and with a runtime warning) (+ add unit tests for deprecation warnings)

File: packages/components/src/navigator/index.tsx (hunk 11 above)
Comment (tyxla) [reply packages/components/src/navigator/index.tsx]: Or we can just exclude the hook from the description?

File: packages/components/src/index.ts (hunk 13 above)
Comment (tyxla) [reply packages/components/src/index.ts:133]: Thanks for confirming 😉 It's a long-going PR, so it never hurts to confirm.

File: packages/components/src/navigator/navigator-back-button/component.tsx (hunk 14 above)
Comment (ciampo) [reply packages/components/src/navigator/navigator-back-button/component.tsx:26]: Nope, that's a bug — well spotted! I'm going to push an update

File: docs/manifest.json (hunk 12 above)
Comment (ciampo) [reply docs/manifest.json]: I think the name is auto-generated from the filename or something else. I like the idea of moving everything under one README, though. We currently have some compound components having one unique README, and other compound components with one README per sub-component. It would be good if we agreed on an approach going forward — @WordPress/gutenberg-components , any preferences?

File: packages/components/src/navigator/index.tsx (hunk 11 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx]: Sure, I can do that

File: packages/components/CHANGELOG.md (hunk 10 above)
Comment (ciampo) [reply packages/components/CHANGELOG.md:22]: I will favour consistency and leave it under the `Enhancements` section, as we've done for other PRs

File: packages/components/src/navigator/navigator-back-button/component.tsx (hunk 14 above)
Comment (ciampo) [reply packages/components/src/navigator/navigator-back-button/component.tsx:26]: Fixed in https://github.com/WordPress/gutenberg/pull/64613/commits/d153bcf0db769fb57c960c90301e09a2ba24c6c2

File: packages/components/src/navigator/index.tsx (hunk 11 above)
Comment (ciampo) [reply packages/components/src/navigator/index.tsx]: Mentions removed in https://github.com/WordPress/gutenberg/pull/64613/commits/3c3df877bd654c92a8898d363c047e06934dc276.

Folks can always refer to [this section](https://github.com/WordPress/gutenberg/blob/3c3df877bd654c92a8898d363c047e06934dc276/packages/components/src/navigator/navigator/README.md#the-navigator-object) in the main README for docs about the `useNavigator` hook (plus, I've added better JSDocs in https://github.com/WordPress/gutenberg/pull/64613/commits/567f41879cd72147f8db95be614c7fafeddf2d6b)

File: docs/manifest.json (hunk 12 above)
Comment (ciampo) [reply docs/manifest.json]: I merged all READMEs into one: https://github.com/WordPress/gutenberg/pull/64613/commits/5e33f26ba86f2308b331a0b7af60feb931836727, https://github.com/WordPress/gutenberg/pull/64613/commits/0cff1065813626fde5a2e14f444e69dbe39c6bfd and https://github.com/WordPress/gutenberg/pull/64613/commits/cea2b8e66c939f20fbdaa931ec501c7991622cf7.

I think I like keeping one README more, but happy to hear other's thoughts before merging the PR.

Comment (mirka) [reply docs/manifest.json]: > I think I like keeping one README more

No strong opinion. I might have an updated opinion after I attempt auto-generating the READMEs 🙈

Comment (tyxla) [reply docs/manifest.json]: I prefer a single README too - primarily because it's a better developer experience. It's way easier to understand the entire component family, rather than having to jump between multiple READMEs and figure out the puzzle on your own.

Comment (ciampo) [reply docs/manifest.json]: Sounds good — I guess we can go with the one README for now and tweak later if necessary, together with the rest of the components that will need similar changes.

**Comment by ciampo** [comment]
All feedback should be addressed, including [this conversation](https://github.com/WordPress/gutenberg/pull/64613#discussion_r1777390024) on keeping one vs many READMEs.

File: packages/components/src/navigator/README.md [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+
+The children elements.
+
+-   Required: Yes
+
+#### `Navigator.Screen`
+
+###### `path`: `string`
+
+The screen&quot;s path, matched against the current path stored in the navigator.
```
Comment (tyxla) [inline packages/components/src/navigator/README.md]:
```suggestion
The screen's path, matched against the current path stored in the navigator.
```

File: packages/components/src/navigator/README.md [hunk 16]
```diff
@@ -0,0 +1,168 @@
+# `Navigator`
+
+`Navigator` is a collection components that allow rendering nested views/panels/menus (via the [`Navigator.Screen` component](/packages/components/src/navigator/navigator-screen/README.md)) and navigate between them (via the [`Navigator.Button`](/packages/components/src/navigator/navigator-button/README.md) and [`Navigator.BackButton`](/packages/components/src/navigator/navigator-back-button/README.md) components).
```
Comment (tyxla) [inline packages/components/src/navigator/README.md]: Didn't we remove these READMEs? 

```suggestion
`Navigator` is a collection of components that allow rendering nested views/panels/menus (via the `Navigator.Screen`) and navigate between them (via the `Navigator.Button` and `Navigator.BackButton` components).
```

File: packages/components/src/navigator/README.md [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+
+-   `focusTargetSelector`: `string`. An optional property used to specify the CSS selector used to restore focus on the matching element when navigating back;
+-   `isBack`: `boolean`. An optional property used to specify whether the navigation should be considered as backwards (thus enabling focus restoration when possible, and causing the animation to be backwards too);
+-   `skipFocus`: `boolean`. An optional property used to opt out of `Navigator`'s focus management, useful when the consumer of the component wants to manage focus themselves;
+
+###### `goBack`: `( path: string, options: NavigateOptions ) => void`
+
+The `goBack` function allows navigating to the parent screen. Parent/child navigation only works if the paths you define are hierarchical (see note above).
+
+When a match is not found, the function will try to recursively navigate the path hierarchy until a matching screen (or the root `/`) are found.
```
Comment (tyxla) [inline packages/components/src/navigator/README.md]:
```suggestion
When a match is not found, the function will try to recursively navigate the path hierarchy until a matching screen (or the root `/`) is found.
```

File: packages/components/src/navigator/README.md [hunk 18]
```diff
@@ -0,0 +1,168 @@
...
+
+The available options are the same as for the `goTo` method, except for the `isBack` property, which is not available for the `goBack` method.
+
+###### `location`: `NavigatorLocation`
+
+The `location` object represent the current location, and has a few properties:
```
Comment (tyxla) [inline packages/components/src/navigator/README.md]:
```suggestion
The `location` object represents the current location, and has a few properties:
```

File: packages/components/src/navigator/README.md [hunk 19]
```diff
@@ -0,0 +1,168 @@
...
+		</Navigator.Screen>
+	</Navigator>
+);
+```
+
+### Hierarchical `path`s
+
+`Navigator` assumes that screens are organized hierarchically according to their `path`, which should follow a URL-like scheme where each path segment starts with and is separated by the `/` character.
+
+`Navigator` will treat "back" navigations as going to the parent screen — it is therefore responsibility of the consumer of the component to create the correct screen hierarchy.
```
Comment (tyxla) [inline packages/components/src/navigator/README.md]:
```suggestion
`Navigator` will treat "back" navigations as going to the parent screen — it is, therefore, the responsibility of the consumer of the component to create the correct screen hierarchy.
```

File: packages/components/src/navigator/README.md [hunk 20]
```diff
@@ -0,0 +1,168 @@
...
+
+#### `Navigator.Screen`
+
+###### `path`: `string`
+
+The screen&quot;s path, matched against the current path stored in the navigator.
+
+`Navigator` assumes that screens are organized hierarchically according to their `path`, which should follow a URL-like scheme where each path segment starts with and is separated by the `/` character.
+
+`Navigator` will treat "back" navigations as going to the parent screen — it is therefore responsibility of the consumer of the component to create the correct screen hierarchy.
```
Comment (tyxla) [inline packages/components/src/navigator/README.md]:
```suggestion
`Navigator` will treat "back" navigations as going to the parent screen — it is, therefore, the responsibility of the consumer of the component to create the correct screen hierarchy.
```

File: packages/components/src/navigator/test/index.tsx [hunk 21]
```diff
@@ -286,30 +286,30 @@ const MyNavigation = ( {
...
 					<p>{ SCREEN_TEXT.invalidHtmlPath }</p>
 					<CustomNavigatorBackButton
 						onClick={ onNavigatorButtonClick }
 					>
 						{ BUTTON_TEXT.back }
 					</CustomNavigatorBackButton>
-				</NavigatorScreen>
+				</Navigator.Screen>
 
 				{ /* A `NavigatorScreen` with `path={ PATHS.NOT_FOUND }` is purposefully not included. */ }
```
Comment (tyxla) [inline packages/components/src/navigator/test/index.tsx]:
```suggestion
				{ /* A `Navigator.Screen` with `path={ PATHS.NOT_FOUND }` is purposefully not included. */ }
```

File: packages/components/src/navigator/stories/index.story.tsx [hunk 22]
```diff
@@ -40,7 +40,7 @@ const meta: Meta< typeof NavigatorProvider > = {
 						 * detail of the Navigator component. Do not use outside of
 						 * its source code.
 						 */
-						[data-wp-component="NavigatorProvider"] {
+						[data-wp-component="Navigator"] {
 							height: 250px;
 						}
 						[data-wp-component="NavigatorScreen"]:not([data-sticky]) {
```
Comment (tyxla) [inline packages/components/src/navigator/stories/index.story.tsx]: I think we need to change this one too:

```suggestion
						[data-wp-component="Navigator.Screen"]:not([data-sticky]) {
```

File: packages/components/src/navigator/types.ts [hunk 23]
```diff
@@ -100,6 +100,24 @@ export type NavigatorProps = {
...
 	/**
 	 * The screen's path, matched against the current path stored in the navigator.
+	 *
+	 * `Navigator` assumes that screens are organized hierarchically according
+	 * to their `path`, which should follow a URL-like scheme where each path
+	 * segment starts with and is separated by the `/` character.
+	 *
+	 * `Navigator` will treat "back" navigations as going to the parent screen —
+	 * it is therefore responsibility of the consumer of the component to create
+	 * the correct screen hierarchy.
```
Comment (tyxla) [inline packages/components/src/navigator/types.ts]:
```suggestion
	 * `Navigator` will treat "back" navigations as going to the parent screen —
	 * it is, therefore, the responsibility of the consumer of the component to
	 * create the correct screen hierarchy.
```

**Review by tyxla — APPROVED** [review]
Looking great, excited to finally stabilize the `Navigator` family 👍 

Left a few minor suggestions, mostly grammar, typos, etc.

Thank you 🚀

File: packages/components/src/navigator/stories/index.story.tsx (hunk 22 above)
Comment (ciampo) [reply packages/components/src/navigator/stories/index.story.tsx]: The current version is correct — we're setting the height on the wrapper, as also suggested in the README. Each screen, if the wrapper's height is set, will follow that height thanks to the grid layout.

File: packages/components/src/navigator/README.md (hunk 16 above)
Comment (ciampo) [reply packages/components/src/navigator/README.md]: 🤦

File: packages/components/src/navigator/stories/index.story.tsx (hunk 22 above)
Comment (tyxla) [reply packages/components/src/navigator/stories/index.story.tsx]: I think you're looking at the wrong place - this is about the padding, and in my testing the padding appears only if we use the dot notation.

Comment (tyxla) [reply packages/components/src/navigator/stories/index.story.tsx]: Alternatively, feel free to remove this style if you think we don't need it. But my point was that if you want the padding, you need to add the dot.

Comment (ciampo) [reply packages/components/src/navigator/stories/index.story.tsx]: You're right — I looked at the wrong diff 😅 

I pushed a fix as you suggested.

File: docs/manifest.json (hunk 3 above)
Comment (ciampo) [reply docs/manifest.json:1117]: Hey @WordPress/wordpress-org-meta-team , could you help with this? We need to update the Block Editor Handbook, removing the following components:

- [`NavigatorProvider`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-provider/)
- [`NavigatorButton`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-button/)
- [`NavigatorBackButton`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-back-button/)
- [`NavigatorScreen`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-screen/)
- [`NavigatorToParentButton`](https://developer.wordpress.org/block-editor/reference-guides/components/navigator-to-parent-button/)


If possible, we'd also like to set a redirect for those pages being removed. The redirect should point to [the `Navigator` component](https://developer.wordpress.org/block-editor/reference-guides/components/navigator/).


Could you please help with that? Thank you 🙏

Comment (adamwoodnz) [reply docs/manifest.json:1117]: These updates have shipped. I noticed the formatting on the new doc loses consistency around the `Navigator.Button` point (heading is not bold):

Screenshot 2024-10-01 at 10 42 06 AM

Comment (ciampo) [reply docs/manifest.json:1117]: Thank you!



> I noticed the formatting on the new doc loses consistency around the `Navigator.Button` point (heading is not bold):

Good catch! Opened https://github.com/WordPress/gutenberg/pull/65763 to fix it
